server/mailet/mailets/src/test/java/org/apache/james/transport/matchers/dlp/DlpTest.java

```diff
@@ -0,0 +1,471 @@
+            .setSubject("=?ISO-8859-1?Q?I_just_bought_a_pon=E9?=")
+            .setText("Meaningless text");
+
+        FakeMail mail = FakeMail
+            .builder()
+            .sender(OTHER_AT_JAMES)
+            .recipient(RECIPIENT2)
+            .mimeMessage(meaninglessText)
+            .build();
+
+        assertThat(dlp.match(mail)).containsOnly(RECIPIENT2);
+    }
+
+    @Test
+    void matchShouldReturnRecipientsWhenRulesMatchesAQuotedPrintableRecipient() throws Exception {
+        Dlp dlp = new Dlp(
+            asRulesLoaderFor(
+                JAMES_APACHE_ORG_DOMAIN,
+                DlpDomainRules.builder()
+                    .recipientRule(Id.of("id1"), Pattern.compile("Benoît"))
+                    .build()));
+
+        MimeMessageBuilder meaninglessText = MimeMessageBuilder
+            .mimeMessageBuilder()
+            .addToRecipient("=?ISO-8859-1?Q?Beno=EEt_TELLIER?=")
+            .setSubject("=?ISO-8859-1?Q?I_just_bought_a_pon=E9?=")
+            .setText("Meaningless text");
+
+        FakeMail mail = FakeMail
+            .builder()
+            .sender(OTHER_AT_JAMES)
+            .recipient(RECIPIENT2)
+            .mimeMessage(meaninglessText)
+            .build();
+
+        assertThat(dlp.match(mail)).containsOnly(RECIPIENT2);
+    }
+
+    @Test
+    void matchShouldReturnRecipientsWhenRulesMatchesAQuotedPrintableSender() throws Exception {
+        Dlp dlp = new Dlp(
+            asRulesLoaderFor(
+                JAMES_APACHE_ORG_DOMAIN,
+                DlpDomainRules.builder()
+                    .senderRule(Id.of("id1"), Pattern.compile("Benoît"))
+                    .build()));
+
+        MimeMessageBuilder meaninglessText = MimeMessageBuilder
+            .mimeMessageBuilder()
+            .addFrom("=?ISO-8859-1?Q?Beno=EEt_TELLIER?=")
+            .setSubject("=?ISO-8859-1?Q?I_just_bought_a_pon=E9?=")
+            .setText("Meaningless text");
+
+        FakeMail mail = FakeMail
+            .builder()
+            .sender(OTHER_AT_JAMES)
+            .recipient(RECIPIENT2)
+            .mimeMessage(meaninglessText)
+            .build();
+
+        assertThat(dlp.match(mail)).containsOnly(RECIPIENT2);
+    }
+
+    @Test
+    void matchShouldAttachMatchingRuleNameToMail() throws Exception {
+        Dlp dlp = new Dlp(
+            asRulesLoaderFor(
+                JAMES_APACHE_ORG_DOMAIN,
+                DlpDomainRules.builder()
+                    .recipientRule(Id.of("should not match recipient"), Pattern.compile(RECIPIENT3.asString()))
+                    .senderRule(Id.of("should match sender"), Pattern.compile(JAMES_APACHE_ORG))
+                    .build()));
+
+        FakeMail mail = FakeMail.builder()
+            .sender(ANY_AT_JAMES)
+            .recipient(RECIPIENT1)
+            .recipient(RECIPIENT2)
+            .build();
+
+        dlp.match(mail);
+
+        assertThat(mail.getAttribute("DlpMatchedRule")).isEqualTo("should match sender");
+    }
+
+}
```

Comment:
can we just add a (eventually failing) test with an encoded body?

Reply:
ok